### PR TITLE
Friendly-name filter properties

### PR DESCRIPTION
**Pull request by Gman98ish**
## Overview

Proposed changes for having friendly filter names

~~Not ready to merge yet, as I haven't had a chance to test it with existing systems~~

### CODE REVIEW

File: src/PSS/RequestClient.php [hunk 1]
```diff
@@ -10,6 +10,16 @@ class RequestClient extends BaseClient
 {
     protected $basePath = 'pss/';
 
+    protected $requestMap = [
```
Comment (MoHangingWithLegends) [inline src/PSS/RequestClient.php]: why cant we just define this in entity

File: src/Entity.php [hunk 2]
```diff
@@ -18,7 +22,9 @@ public function __construct($attributes = [])
             $attributes = (array) $attributes;
         }
         
-        $this->attributes = $attributes;
+        foreach ($attributes as $name => $value) {
```
Comment (MoHangingWithLegends) [inline src/Entity.php]: just use fill like i did

File: src/Client.php [hunk 3]
```diff
@@ -243,4 +243,43 @@ protected function getUserAgent()
             'php/'.PHP_MAJOR_VERSION.'.'.PHP_MINOR_VERSION
         ]);
     }
+
+    /**
+     * @param array|Entity $item
+     * @param array $map
+     * @return array
+     */
+    public function apiToFriendly($item, $map)
```
Comment (MoHangingWithLegends) [inline src/Client.php]: better to call these snake to camel

Comment (Gman98ish) [reply src/Client.php]: They aren't always snake to camel, but they are always api to friendly :)

File: src/PSS/RequestClient.php (hunk 1 above)
Comment (Gman98ish) [reply src/PSS/RequestClient.php]: I think we want to move all API specific knowledge out of the entity classes

/cc @gavtaylor

**Comment by Gman98ish** [comment]
@jbirchevans I've made a few tweaks, would you be able to re-approve, please?

File: src/PSS/RequestClient.php (hunk 1 above)
Comment (phily245) [reply src/PSS/RequestClient.php]: How would this look if one client interacted with multiple entities?

Comment (Gman98ish) [reply src/PSS/RequestClient.php]: You'd have multiple map properties

Comment (Gman98ish) [reply src/PSS/RequestClient.php]: We could move this into the Entity as a constant, I guess. Main issue of doing it this way is if an entity is used in multiple clients

**Pull request merged**